src/Elastic.Apm/DiagnosticListeners/HttpDiagnosticListenerImplBase.cs

```diff
@@ -118,6 +119,14 @@ protected override void HandleOnNext(KeyValuePair<string, object> kv)
 
 		private void ProcessStartEvent(TRequest request, Uri requestUrl)
 		{
+			if (_realAgent?.TracerInternal.CurrentSpan is Span currentSpan)
+			{
+				// if there's a current span that has been instrumented for Azure, don't create a span for
```

Comment:
👍 nice, this is definitely needed to avoid double span creation in combination with the Azure diagnostic source listeners.

One consequence here: In `ProcessStopEvent` we do `ProcessingRequests.TryRemove(request, out var span)` and print a warning if it returns `false`. That warning needs to be adapted because with this it can return `false` and that's ok.